File: Difficulty: Easy/Two Sum - Pair with Given Sum/two-sum-pair-with-given-sum.java
class Solution {
    boolean twoSum(int arr[], int target) {
        HashSet<Integer> seen=new HashSet<>();
        for(int i=0;i<arr.length;i++) {
            int required=target-arr[i];
            if(seen.contains(required)) {
                return true;
            }
            seen.add(arr[i]);
        }
        return false;
    }
}

File: Difficulty: Easy/Two Sum - Pair with Given Sum/two-sum-pair-with-given-sum.py
class Solution:
	def twoSum(self, arr, target):
		n=len(arr)
		left=0
		right=n-1
		arr.sort()
		while left<right:
		    if arr[left]+arr[right]==target:
		        return True
		    elif arr[left]+arr[right]>target:
		        right-=1 
		    else:
		        left+=1
		return False